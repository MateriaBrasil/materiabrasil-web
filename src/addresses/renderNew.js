import React from 'react'

import Form from './Form'
import createWithParams from './createWithParams'

export default props => ({ create, creating, error }) => {
  const { id } = props.match.params

  return (
    <Form
      {...props}
      onSubmit={createWithParams(create, id)}
      creating={creating}
      createError={error}
    />
  )
}
