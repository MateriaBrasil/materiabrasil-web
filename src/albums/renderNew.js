import React from 'react'
import { New } from 'croods'
import { Formik } from 'formik'
import Button from '@material-ui/core/Button'
import Dialog from 'materials/Dialog'
import Input from '@material-ui/core/Input'
import InputLabel from '@material-ui/core/InputLabel'
import FormHelperText from '@material-ui/core/FormHelperText'
import Typography from '@material-ui/core/Typography'
import FormControl from '@material-ui/core/FormControl'

export default props => routeProps => {
  const validate = values => {
    let errors = {}

    if (!values.name) {
      errors.name = 'Required'
    }

    return errors
  }
  const { history } = routeProps
  const { setCurrentUser, currentUser } = props

  return (
    <Dialog {...props} {...routeProps} title="Adicionar novo álbum">
      {currentUser && (
        <New
          name="albums"
          render={({ create }) => (
            <Formik
              initialValues={{ name: '' }}
              validate={validate}
              onSubmit={values => {
                create({
                  name: values.name,
                })
              }}
              render={({
                values,
                errors,
                touched,
                handleChange,
                handleBlur,
                handleSubmit,
              }) => (
                <form onSubmit={handleSubmit}>
                  <FormControl
                    error={touched['name'] && !!errors['name']}
                    aria-describedby="name-error-text"
                    fullWidth
                  >
                    <InputLabel htmlFor="name-error">Nome do album</InputLabel>
                    <Input
                      id="name"
                      name="name"
                      onChange={handleChange}
                      onBlur={handleBlur}
                      value={values.name}
                    />
                    <FormHelperText id="name-error-text">
                      {touched['name'] && errors['name']}
                    </FormHelperText>
                  </FormControl>
                  <Button type="submit">Adicionar</Button>
                </form>
              )}
            />
          )}
          renderCreated={album => {
            setCurrentUser({
              ...currentUser,
              albums: [...currentUser.albums, album],
            })
            history.push(`/materials/${routeProps.match.params.id}/albums`)
          }}
        />
      )}
      {!props.currentUser && (
        <Typography color="inherit">
          Você não está logado, se logue para continuar
        </Typography>
      )}
    </Dialog>
  )
}
