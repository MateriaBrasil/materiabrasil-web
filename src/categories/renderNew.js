import React from 'react'
import FormGroup from '@material-ui/core/FormGroup'

import renderCategory from './renderCategory'

export default parentProps => ({ create, creating }) => {
  const props = { ...parentProps, create, creating }
  const { categories } = props

  return (
    <FormGroup>
      {categories.map(category =>
        renderCategory({ ...props, rootCategory: category })(category),
      )}
    </FormGroup>
  )
}
