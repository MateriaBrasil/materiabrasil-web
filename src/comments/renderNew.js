import React from 'react'

import Form from './Form'
import createWithParams from './createWithParams'

export default props => ({ create, creating, error }) => {
  const { id } = props

  return (
    <Form
      onSubmit={createWithParams(create, id)}
      submitting={creating}
      createError={error}
      {...props}
    />
  )
}
