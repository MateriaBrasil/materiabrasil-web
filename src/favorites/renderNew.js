import React from 'react'
import BookmarkIcon from '@material-ui/icons/Bookmark'
import Button from '@material-ui/core/Button'

import createWithParams from './createWithParams'

export default id => ({ create, creating, error }) => (
  <Button
    variant="raised"
    color="secondary"
    disabled={creating}
    onClick={createWithParams(create, id)}
  >
    <BookmarkIcon />
    Salvar
  </Button>
)
