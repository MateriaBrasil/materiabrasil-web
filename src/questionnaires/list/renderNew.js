import React from 'react'
import List from './List'

export default props => ({ create }) => {
  const { questionnaires } = props
  return <List {...props} create={create} list={questionnaires} />
}
