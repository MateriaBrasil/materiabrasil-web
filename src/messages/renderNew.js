import React from 'react'
import { New } from 'croods'

import renderNew from './new/render'
import renderCreated from './new/renderCreated'

export default props => routeProps => (
  <New
    name="messages"
    render={renderNew({ ...props, ...routeProps })}
    renderCreated={renderCreated(routeProps)}
  />
)
