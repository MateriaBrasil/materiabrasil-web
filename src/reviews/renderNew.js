import React from 'react'
import { New } from 'croods'

import reload from '../materials/reload'
import renderNew from './new/render'

export default props => routeProps => (
  <New
    name="reviews"
    render={renderNew({ ...props, ...routeProps })}
    renderCreated={reload({ ...routeProps })}
  />
)
