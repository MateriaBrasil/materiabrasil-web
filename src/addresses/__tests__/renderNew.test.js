import React from 'react'
import renderer from 'react-test-renderer'

import renderNew from '../renderNew'

jest.mock('../Form', () => props => (
  <div {...props}>Form - {props.children}</div>
))

it('renders correctly', () => {
  const createProps = { create: jest.fn() }
  const props = { foo: 'bar', id: 123, match: { params: { id: 123 } } }

  const tree = renderer.create(renderNew(props)(createProps)).toJSON()
  expect(tree).toMatchSnapshot()
})
