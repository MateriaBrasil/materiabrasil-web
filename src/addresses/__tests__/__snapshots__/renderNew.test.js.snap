// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  createError={undefined}
  creating={undefined}
  foo="bar"
  id={123}
  onSubmit={[Function]}
>
  Form - 
</div>
`;
